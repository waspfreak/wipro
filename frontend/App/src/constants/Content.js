export const content = {
  home: {
    title: 'Landing topics screen',
    teams: 'Teams',
    topics: 'Topics',
    hq: 'HQ'
  },
  noticePanel: {
    title: 'A message from One Web',
    message:
      'This project is primarily designed for a mobile experience leveraging React Native however a web design will be created as the project evolves leveraging other web technologies This project is primairly designed for a mobile experience leveraging React Native however a web design will be created as the project evolves leveraging other web technologies'
  },
  trash: {
    title: 'Empty this from the trash',
    message:
      'The item below will be permanently erased in 25 days, or when you empty the trash, whichever comes first. Its data will remain on our internal systems and backups for up to 30 days before it’s completely destroyed.'
  },
  newProject: {
    title: 'Make a topic',
    subtitle:
      'Use projects for anything you’re working on,  like a marketing, a new product launch, or a project for a client.'
  },
  newTeam: {
    title: 'Make a team',
    subtitle:
      'Teams help people with similar roles or interests share things and stay coordinated - like a Sales, Marketing, or Customer Support team'
  }
};
