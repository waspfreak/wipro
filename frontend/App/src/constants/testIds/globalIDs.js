export const globalIds = {
  backButton: 'backButton',
  nextButton: 'nextButton',
  saveButton: 'saveButton',
  saveButtonDisabled: 'saveButtonDisabled',
  menuButton: 'menuButton',
  menuItemOne: 'menuItemOne',
  menuButtonDisabled: 'menuButtonDisabled',
  menuBody: 'menuBody'
};
