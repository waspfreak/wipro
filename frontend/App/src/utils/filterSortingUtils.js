/* filters an array of objects by key value
 * eg. returns array of objects that have key value present */

export const filterByKeyValue = (dataToFilter, filterKey, filterValue) => {
  return !Array.isArray(dataToFilter)
    ? []
    : dataToFilter.filter(item => item[filterKey] === filterValue);
};

/* filters an array of objects by key value
 * then searches value for matching search string */

export const keyValueMatch = (dataToFilter, filterKey, filterValue) => {
  const searchForStringRegex = new RegExp(
    filterValue.replace(/[.*+?^${}()|[\]\\]/g, '\\$&'),
    'gi'
  );

  return !Array.isArray(dataToFilter)
    ? []
    : dataToFilter.filter(item => item[filterKey].match(searchForStringRegex));
};

/* sorts an array of objects by key value alphabetically
 * eg. returns array of objects in alphabetical order of title key value */

export const sortAlphabeticallyByKeyValue = (dataToSort, filterType) => {
  return dataToSort.sort((a, b) =>
    a[filterType].localeCompare(b[filterType], 'en', { sensitivity: 'base' })
  );
};
