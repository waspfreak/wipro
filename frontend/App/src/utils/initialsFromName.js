/* returns initials when given first and last names */

export const initialsFromName = name => {
  const nameArray = name.split(' ');
  const names = [nameArray[0]];
  if (nameArray.length > 1) {
    names.push(nameArray[nameArray.length - 1]);
  }
  let initials = '';
  names.forEach(name => {
    initials += name.charAt(0);
  });

  return initials.toUpperCase();
};
