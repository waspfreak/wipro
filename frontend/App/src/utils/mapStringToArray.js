/* generates a hash from a given string and maps it to a value
in an array.  Mapping item should be the same every time for a particular string
i.e initials
 */

export const mapStringToArray = (str, arr) => {
  return arr[getHashFromString(str) % arr.length];
};

export const getHashFromString = str => {
  let hash = 0;
  let i;
  let chr;

  if (str.length === 0) return hash;
  for (i = 0; i < str.length; i++) {
    chr = str.charCodeAt(i);
    hash = (hash << 5) - hash + chr;
    hash |= 0; // Convert to 32bit integer
  }
  return Math.abs(hash);
};
