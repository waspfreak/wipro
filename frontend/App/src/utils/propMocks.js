export const navigationStatePropsMock = {
  routes: [],
  index: 0,
  isTransitioning: false,
  key: '',
  params: {
    test: 1
  }
};

export const navigationPropMock = {
  dangerouslyGetParent: jest.fn(),
  state: navigationStatePropsMock,
  dispatch: jest.fn(),
  goBack: jest.fn(),
  dismiss: jest.fn(),
  navigate: jest.fn(),
  openDrawer: jest.fn(),
  closeDrawer: jest.fn(),
  toggleDrawer: jest.fn(),
  getParam: jest.fn(),
  setParams: jest.fn(),
  addListener: jest.fn(),
  push: jest.fn(),
  replace: jest.fn(),
  pop: jest.fn(),
  popToTop: jest.fn(),
  isFocused: jest.fn(),
  reset: jest.fn(),
  isFirstRouteInParent: jest.fn()
};
