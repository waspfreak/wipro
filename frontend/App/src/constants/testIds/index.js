import { globalIds } from './globalIDs';
import { topicIds } from './topicIds';

export const testIds = {
  ...globalIds,
  ...topicIds
};
