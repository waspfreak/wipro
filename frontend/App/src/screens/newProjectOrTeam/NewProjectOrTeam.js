import React from 'react';
import { View } from 'react-native';
import { Header } from '../../components/header/Header';
import ButtonMakeProject from '../../components/buttonMakeProject/ButtonMakeProject';
import ButtonIcon from '../../components/buttonIcon/ButtonIcon';
import { SafeAreaView } from 'react-navigation';
import styles from './styles';
import { colors } from '../../styles';
import { content } from '../../constants/Content';
import { testIds } from '../../constants/testIds';

export const NewProjectOrTeam = ({ navigation }) => {
  const leftMenu = () => {
    return (
      <ButtonIcon
        onPress={() => navigation.goBack()}
        color="transparent"
        icon="LeftChevron"
        iconColor={colors.blue}
        iconSizeWidth="28"
        iconSizeHeight="28"
        marginLeft={16}
        testIdentifier={testIds.backButton}
      />
    );
  };
  return (
    <SafeAreaView style={styles.container}>
      <Header title="New Project or Team" left={leftMenu()} />
      <View style={styles.mainContent}>
        <ButtonMakeProject
          onPress={() => navigation.navigate('MakeTopic')}
          title={content.newProject.title}
          subtitle={content.newProject.subtitle}
          testIdentifier="makeTopicButton"
        />
        <ButtonMakeProject
          title={content.newTeam.title}
          subtitle={content.newTeam.subtitle}
        />
      </View>
    </SafeAreaView>
  );
};

export default NewProjectOrTeam;

NewProjectOrTeam.navigationOptions = {
  header: null
};
