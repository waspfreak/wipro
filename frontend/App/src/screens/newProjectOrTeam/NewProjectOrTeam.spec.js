import React from 'react';
import { fireEvent, render } from '@testing-library/react-native';
import NewProjectOrTeam from './NewProjectOrTeam';
import { navigationPropMock } from '../../utils/propMocks';
import { testIds } from '../../constants/testIds';

describe('New Project or Team tests', () => {
  let props;
  let component;

  beforeEach(() => {
    props = {
      navigation: navigationPropMock
    };

    component = render(<NewProjectOrTeam {...props} />);
  });

  afterEach(() => {
    jest.clearAllMocks();
  });
  it('should match NewProjectOrTeam snapshot', () => {
    const { baseElement } = component;
    expect(baseElement).toMatchSnapshot();
  });
  it('should call navigate back when back button is pressed', () => {
    const { getByTestId } = component;
    const backChevron = getByTestId(testIds.backButton);
    fireEvent.press(backChevron);
    expect(navigationPropMock.goBack).toHaveBeenCalled();
  });
  it('should call navigate to go to topics when make topic button is pressed', () => {
    const { getByTestId } = component;
    const makeTopicButton = getByTestId('makeTopicButton');
    fireEvent.press(makeTopicButton);
    expect(navigationPropMock.navigate).toHaveBeenCalledWith('MakeTopic');
  });
});
