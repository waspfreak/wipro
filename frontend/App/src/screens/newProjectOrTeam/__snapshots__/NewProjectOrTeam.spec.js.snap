// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`New Project or Team tests should match NewProjectOrTeam snapshot 1`] = `
<View
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    collapsable={true}
    pointerEvents="box-none"
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      pointerEvents="box-none"
      style={
        Object {
          "paddingBottom": 0,
          "paddingLeft": 0,
          "paddingRight": 0,
          "paddingTop": 20,
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#FFFFFF",
              "borderBottomColor": "#c6c6c6",
              "borderBottomWidth": 1,
              "flexDirection": "row",
              "flexWrap": "wrap",
              "height": 50,
              "justifyContent": "center",
            },
            undefined,
          ]
        }
      >
        <View
          style={
            Object {
              "left": 0,
              "position": "absolute",
            }
          }
        >
          <TouchableHighlight
            activeOpacity={0.85}
            delayPressOut={100}
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "borderRadius": 50,
                  "height": 28,
                  "justifyContent": "center",
                  "width": 28,
                },
                Object {
                  "backgroundColor": "transparent",
                },
                Object {
                  "marginLeft": 16,
                },
                undefined,
              ]
            }
            testID="backButton"
            underlayColor="#3478f6"
          >
            <View>
              <RNSVGSvgView
                align="xMidYMid"
                bbHeight="28"
                bbWidth="28"
                focusable={false}
                height="28"
                meetOrSlice={0}
                minX={0}
                minY={0}
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                      "borderWidth": 0,
                    },
                    undefined,
                    Object {
                      "opacity": 1,
                    },
                    Object {
                      "flex": 0,
                      "height": 28,
                      "width": 28,
                    },
                  ]
                }
                vbHeight={28}
                vbWidth={28}
                width="28"
              >
                <RNSVGGroup
                  fill={
                    Array [
                      0,
                      4278190080,
                    ]
                  }
                  fillOpacity={1}
                  fillRule={1}
                  font={Object {}}
                  matrix={
                    Array [
                      1,
                      0,
                      0,
                      1,
                      0,
                      0,
                    ]
                  }
                  opacity={1}
                  originX={0}
                  originY={0}
                  propList={Array []}
                  rotation={0}
                  scaleX={1}
                  scaleY={1}
                  skewX={0}
                  skewY={0}
                  stroke={null}
                  strokeDasharray={null}
                  strokeDashoffset={null}
                  strokeLinecap={0}
                  strokeLinejoin={0}
                  strokeMiterlimit={4}
                  strokeOpacity={1}
                  strokeWidth={1}
                  vectorEffect={0}
                  x={0}
                  y={0}
                >
                  <RNSVGPath
                    d="M8.292 15.213a1.016 1.016 0 0 1 0-1.426l9.245-9.36a1.431 1.431 0 0 1 2.04 0c.563.57.563 1.496 0 2.066L11.67 14.5l7.909 8.005a1.476 1.476 0 0 1 0 2.066c-.564.57-1.479.57-2.04 0l-9.246-9.36z"
                    fill={
                      Array [
                        0,
                        4279986889,
                      ]
                    }
                    fillOpacity={1}
                    fillRule={0}
                    matrix={
                      Array [
                        1,
                        0,
                        0,
                        1,
                        0,
                        0,
                      ]
                    }
                    opacity={1}
                    originX={0}
                    originY={0}
                    propList={
                      Array [
                        "fill",
                        "fillRule",
                      ]
                    }
                    rotation={0}
                    scaleX={1}
                    scaleY={1}
                    skewX={0}
                    skewY={0}
                    stroke={null}
                    strokeDasharray={null}
                    strokeDashoffset={null}
                    strokeLinecap={0}
                    strokeLinejoin={0}
                    strokeMiterlimit={4}
                    strokeOpacity={1}
                    strokeWidth={1}
                    vectorEffect={0}
                    x={0}
                    y={0}
                  />
                </RNSVGGroup>
              </RNSVGSvgView>
            </View>
          </TouchableHighlight>
        </View>
        <Text
          style={
            Object {
              "color": "#000000",
              "fontFamily": "SFProDisplay-Bold",
              "fontSize": 17,
            }
          }
        >
          New Project or Team
        </Text>
      </View>
      <View
        style={
          Object {
            "paddingHorizontal": 20,
            "paddingVertical": 40,
          }
        }
      >
        <TouchableOpacity
          activeOpacity={0.2}
          testID="makeTopicButton"
        >
          <View
            style={
              Object {
                "alignContent": "center",
                "alignItems": "center",
                "alignSelf": "center",
                "borderColor": "#c6c6c6",
                "borderWidth": 1,
                "height": 201,
                "marginVertical": 5,
                "paddingHorizontal": 16,
                "paddingVertical": 12,
                "textAlign": "center",
                "width": "90%",
              }
            }
          >
            <View
              style={
                Object {
                  "alignContent": "center",
                  "alignItems": "center",
                  "backgroundColor": "#5856d6",
                  "borderRadius": 99999,
                  "flexDirection": "row",
                  "height": 40,
                  "justifyContent": "center",
                  "marginBottom": 9,
                  "marginTop": 14,
                  "width": 40,
                }
              }
            />
            <Text
              style={
                Object {
                  "color": "#000000",
                  "fontFamily": "SFProDisplay-Bold",
                  "fontSize": 20,
                  "marginBottom": 10,
                }
              }
            >
               
              Make a topic
            </Text>
            <Text
              style={
                Object {
                  "color": "#000000",
                  "fontFamily": "SFProDisplay-Regular",
                  "fontSize": 13,
                  "lineHeight": 18,
                  "textAlign": "center",
                }
              }
            >
              Use projects for anything youâ€™re working on,  like a marketing, a new product launch, or a project for a client.
            </Text>
          </View>
        </TouchableOpacity>
        <TouchableOpacity
          activeOpacity={0.2}
        >
          <View
            style={
              Object {
                "alignContent": "center",
                "alignItems": "center",
                "alignSelf": "center",
                "borderColor": "#c6c6c6",
                "borderWidth": 1,
                "height": 201,
                "marginVertical": 5,
                "paddingHorizontal": 16,
                "paddingVertical": 12,
                "textAlign": "center",
                "width": "90%",
              }
            }
          >
            <View
              style={
                Object {
                  "alignContent": "center",
                  "alignItems": "center",
                  "backgroundColor": "#5856d6",
                  "borderRadius": 99999,
                  "flexDirection": "row",
                  "height": 40,
                  "justifyContent": "center",
                  "marginBottom": 9,
                  "marginTop": 14,
                  "width": 40,
                }
              }
            />
            <Text
              style={
                Object {
                  "color": "#000000",
                  "fontFamily": "SFProDisplay-Bold",
                  "fontSize": 20,
                  "marginBottom": 10,
                }
              }
            >
               
              Make a team
            </Text>
            <Text
              style={
                Object {
                  "color": "#000000",
                  "fontFamily": "SFProDisplay-Regular",
                  "fontSize": 13,
                  "lineHeight": 18,
                  "textAlign": "center",
                }
              }
            >
              Teams help people with similar roles or interests share things and stay coordinated - like a Sales, Marketing, or Customer Support team
            </Text>
          </View>
        </TouchableOpacity>
      </View>
    </View>
  </View>
</View>
`;
